docs/contributing/MODEL_MANAGER.md

```diff
@@ -0,0 +1,1214 @@
+
+config = InvokeAIAppConfig.get_config()
+db_conn = sqlite3.connect(config.db_path.as_posix(), check_same_thread=False)
+store = ModelRecordServiceBase.open(config, db_conn, lock)
+```
+
+_A note on simultaneous access to `invokeai.db`_: The current InvokeAI
+service architecture for the image and graph databases is careful to
+use a shared sqlite3 connection and a thread lock to ensure that two
+threads don't attempt to access the database simultaneously. However,
+the default `sqlite3` library used by Python reports using
+**Serialized** mode, which allows multiple threads to access the
+database simultaneously using multiple database connections (see
+https://www.sqlite.org/threadsafe.html and
+https://ricardoanderegg.com/posts/python-sqlite-thread-safety/). Therefore
+it should be safe to allow the record service to open its own SQLite
+database connection. Opening a model record service should then be as
+simple as `ModelRecordServiceBase.open(config)`.
+
+### Fetching a Model's Configuration from `ModelRecordServiceBase`
+
+Configurations can be retrieved in several ways.
+
+#### get_model(key) -> AnyModelConfig:
+
+The basic functionality is to call the record store object's
+`get_model()` method with the desired model's unique key. It returns
+the appropriate subclass of ModelConfigBase:
+
+```
+model_conf = store.get_model('f13dd932c0c35c22dcb8d6cda4203764')
+print(model_conf.path)
+
+>> '/tmp/models/ckpts/v1-5-pruned-emaonly.safetensors'
+
+```
+
+If the key is unrecognized, this call raises an
+`UnknownModelException`.
+
+#### exists(key) -> AnyModelConfig:
+
+Returns True if a model with the given key exists in the databsae.
+
+#### search_by_path(path) -> AnyModelConfig:
+
+Returns the configuration of the model whose path is `path`. The path
+is matched using a simple string comparison and won't correctly match
+models referred to by different paths (e.g. using symbolic links).
+
+#### search_by_name(name, base, type) -> List[AnyModelConfig]:
+
+This method searches for models that match some combination of `name`,
+`BaseType` and `ModelType`. Calling without any arguments will return
+all the models in the database.
+
+#### all_models() -> List[AnyModelConfig]:
+
+Return all the model configs in the database. Exactly equivalent to
+calling `search_by_name()` with no arguments.
+
+#### search_by_tag(tags) -> List[AnyModelConfig]:
+
+`tags` is a list of strings. This method returns a list of model
+configs that contain all of the given tags. Examples:
+
+```
+# find all models that are marked as both SFW and as generating
+# background scenery
+configs = store.search_by_tag(['sfw', 'scenery'])
+```
+
+Note that only tags are not searchable in this way. Other fields can
+be searched using a filter:
+
+```
+commercializable_models = [x for x in store.all_models() \
+                           if x.license.contains('allowCommercialUse=Sell')]
+```
+
+#### version() -> str:
+
+Returns the version of the database, currently at `3.2`
```

Comment:
I don't think the `ModelRecordServiceBase` should be responsible for surfacing the DB version.